src/EPR.SubsidiaryBulkUpload.Application/Extensions/UserRequestModelExtensions.cs

```diff
@@ -0,0 +1,27 @@
+﻿using EPR.SubsidiaryBulkUpload.Application.Models;
+
+namespace EPR.SubsidiaryBulkUpload.Application.Extensions;
+
+public static class UserRequestModelExtensions
+{
+    public static UserRequestModel ToUserRequestModel(this IDictionary<string, string> metadata)
+    {
+        if (metadata is null)
+        {
+            return null;
+        }
+
+        var caseInsensitiveMetadata = new Dictionary<string, string>(metadata, StringComparer.OrdinalIgnoreCase);
+
+        if (!caseInsensitiveMetadata.ContainsKey("UserId") || !caseInsensitiveMetadata.ContainsKey("OrganisationId"))
```

Comment:
Add condition that the user ID is a Guid here (Guid.TryParse)

Reply:
Done.